lib/practice_exercises.rb

```diff
@@ -1,13 +1,45 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n) -> time depends on how long the string is.
+              #string of 3 chars will take less time than one of 100000 chars
+# Space Complexity: O(1) -> nothing new being created
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+  i = 0
+  array = string.upcase.split('')
+
+  punctuation = %w[. , / \ " ; : ! & ^ # $ @ % ( ) \ ]
+
+  punctuation.each do |i|
+    if array.include?(i)
+      array.delete(i)
+    end
+  end
+
+  while i <= array.length
+    return false if array[i] != array[-(i+1)]
+    i += 1
+  end 
+
+  return true
+  
 end
 
 # Time Complexity: ?
 # Space Complexity: ?
 def longest_prefix(strings)
-  raise NotImplementedError, "Not implemented yet"
+  shortest_string = strings.min_by(&:length)
+
+  string_length = shortest_string.length
+  x = string_length - 1
+  
+  strings.each do |string|
+    string_length.times do
+      if !string.include?(shortest_string[0..x])
```

Comment:
This `.include?` involves a loop which runs x times, which is at max, m, the number of letters in a string.